### PR TITLE
feat: add ungrammar-languageserver support

### PR DESCRIPTION
**Pull request by binhtran432k**
Add [ungrammar-languageserver](https://github.com/binhtran432k/ungrammar-language-features) support

### CODE REVIEW

File: lua/lspconfig/server_configurations/ungrammar_languageserver.lua [hunk 1]
```diff
@@ -0,0 +1,36 @@
+local util = require 'lspconfig.util'
+
+return {
+  default_config = {
+    cmd = { 'ungrammar-languageserver', '--stdio' },
+    filetypes = { 'ungrammar' },
+    root_dir = function(fname)
+      return util.root_pattern '.git'(fname)
+    end,
```
Comment (glepnir) [inline lua/lspconfig/server_configurations/ungrammar_languageserver.lua:9]:
```suggestion
    root_dir = util.find_git_ancestor, 
```

Comment (binhtran432k) [reply lua/lspconfig/server_configurations/ungrammar_languageserver.lua]: > 

Thank @glepnir, I have force-pushed to avoid the noise commits.

Comment (justinmk) [reply lua/lspconfig/server_configurations/ungrammar_languageserver.lua]: don't we want to discourage `find_git_ancestor` ?

Comment (glepnir) [reply lua/lspconfig/server_configurations/ungrammar_languageserver.lua]: can be handled uniformly during subsequent cleanup. this is relatively simple at present.  plann to stop recommending this function syntax sugar for all new PRs now?

Comment (glepnir) [reply lua/lspconfig/server_configurations/ungrammar_languageserver.lua]: I have create #3313 then there can use `root_dir = vim.fs.root(0, '.git')`

**Comment by justinmk** [comment]
needs a rebase, the `server_configurations/` directory was renamed to `configs/`  https://github.com/neovim/nvim-lspconfig/pull/3330

**Pull request merged**